File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {

		Stack<Integer> stk = new Stack<>();

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String s = br.readLine();
		int n = Integer.parseInt(s);

		for (int i = 0; i < n; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			String cmd = st.nextToken();

			if (cmd.equals("push")) {
				int num = Integer.parseInt(st.nextToken());

				stk.push(num);
			} else if (cmd.equals("pop")) {
				StringBuilder sb = new StringBuilder();
				if (stk.empty()) {
					sb.append("-1");
				} else {
					sb.append(stk.peek());
					stk.pop();
				}
				System.out.println(sb);
			} else if (cmd.equals("size")) {
				StringBuilder sb = new StringBuilder();

				sb.append(stk.size());
				System.out.println(sb);
			} else if (cmd.equals("empty")) {
				StringBuilder sb = new StringBuilder();

				if (stk.empty())
					sb.append("1");
				else
					sb.append("0");

				System.out.println(sb);
			} else if (cmd.equals("top")) {
				StringBuilder sb = new StringBuilder();

				if (stk.empty()) {
					sb.append("-1");
				} else {
					sb.append(stk.peek());
				}

				System.out.println(sb);
			}
		}
	}
}

File: 백준/Silver/10828. 스택/스택.py
n = int(input())
st = []

for _ in range(0, n):
    cmd = input().split()
    s = cmd[0]
    if s == "push":
        num = int(cmd[1])
        st.append(num)
    elif s == "pop":
        if len(st) == 0:
            print("-1")
        else:
            print(st[-1])
            st.pop()
    elif s == "size":
        print(len(st))
    elif s == "empty":
        if len(st) == 0:
            print(1)
        else:
            print(0)
    else:
        if len(st) == 0:
            print("-1")
        else:
            print(st[-1])
